### ISSUE TITLE
Variable Assignment Error

### ISSUE THREAD

**Issue by jeffwcx** [sass/libsass#2551]
# Variable Assignment Error 

## input.scss

```scss
@function test () {
  $result: 1;
  @for $i from 1 through 10 {
    @if $i % 2 != 0 {
      $result: $i;
    }
  }
  @return $result;
}
@debug "result: " + test();
```

## Actual results

[3.5.0.beta.2][1]
```
result: 10
```

## Expected result

ruby sass 3.5.5
```css
result: 9
```

version info:
```cmd
node-sass       4.7.2   (Wrapper)       [JavaScript]
libsass         3.5.0.beta.2    (Sass Compiler) [C/C++]
```

[todo]: # (Go to http://libsass.ocbnet.ch/srcmap)
[todo]: # (Enter your SCSS code and hit compile)
[todo]: # (Click `bookmark` and replace the url)
[todo]: # (link is used in actual results above)

[1]: http://libsass.ocbnet.ch/srcmap/#QGZ1bmN0aW9uIHRlc3QgKCkgewogICRyZXN1bHQ6IDE7CiAgQGZvciAkaSBmcm9tIDEgdGhyb3VnaCAxMCB7CiAgICBAaWYgJGkgJSAyICE9IDAgewogICAgICAkcmVzdWx0OiAkaTsKICAgIH0KICB9CiAgQHJldHVybiAkcmVzdWx0Owp9CkBkZWJ1ZyAicmVzdWx0OiAiICsgdGVzdCgpOw==

**Comment by mgreter** [comment]
I can't really [reproduce the issue][1]!?

[1]: http://libsass.ocbnet.ch/srcmap/#QGZ1bmN0aW9uIHRlc3QgKCkgewogICRyZXN1bHQ6IDE7CiAgQGZvciAkaSBmcm9tIDEgdGhyb3VnaCAxMCB7CiAgICBAaWYgJGkgJSAyICE9IDAgewogICAgICAkcmVzdWx0OiAkaTsKICAgIH0KICB9CiAgQHJldHVybiAkcmVzdWx0Owp9CmZvbyB7CiAgcmVzdWx0OiB0ZXN0KCk7Cn0=

**Comment by jeffwcx** [comment]
@mgreter , this issue may have already fixed in #2330 . I down grade node-sass's version to 4.1.1. and it works.  Thank for your answer.
